### PR TITLE
Update CHANGELOG for Horizon release 2.31.0

### PR DESCRIPTION
**Pull request by aditya1702**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://developers.stellar.org/api/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Update Changelog for the release.

### Why

N/A

### Known limitations

N/A

### CODE REVIEW

**Review by tamirms — APPROVED** [review]

**Review by sreuland — APPROVED** [review]
lgtm

**Comment by urvisavla** [comment]
I think we should mention in the changelog that EXPERIMENTAL_BUCKETLIST_DB=true must be set in the captive-core config. https://github.com/stellar/go/commit/89068a97ead78fbce9168f53ce6898a866b85815.

**Comment by sreuland** [comment]
> I think we should mention in the changelog that EXPERIMENTAL_BUCKETLIST_DB=true must be set in the captive-core config. [89068a9](https://github.com/stellar/go/commit/89068a97ead78fbce9168f53ce6898a866b85815).

related, do we need to then also include the opposite use case:
> set EXPERIMENTAL_BUCKETLIST_DB=false if setting Horizon CAPTIVE_CORE_USE_DB=false

File: services/horizon/CHANGELOG.md [hunk 1]
```diff
@@ -3,12 +3,26 @@
 All notable changes to this project will be documented in this
 file. This project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 2.31.0
+
+As part of Protocol 21, if you are using a custom captive-core config, make sure to set `EXPERIMENTAL_BUCKETLIST_DB=true` there along with `CAPTIVE_CORE_USE_DB=true` in the Horizon config. If you are setting `CAPTIVE_CORE_USE_DB=false`, ensure the bucketlist parameter to false too.
```
Comment (urvisavla) [inline services/horizon/CHANGELOG.md:8]: Isn't CAPTIVE_CORE_USE_DB set to true by default and setting it to false is *not* recommended and comes with a deprecation warning? I'm not sure if including this suggests that it's okay to use CAPTIVE_CORE_USE_DB=false. What do you think @sreuland?

Comment (sreuland) [inline services/horizon/CHANGELOG.md:8]:
```suggestion
If you are using a custom captive-core config and setting Horizon `CAPTIVE_CORE_USE_DB=false`, then need to make sure to set `EXPERIMENTAL_BUCKETLIST_DB=false` in the captive core config toml to be compatible with stellar-core versions 21 and earlier
```

Comment (sreuland) [reply services/horizon/CHANGELOG.md:8]: @urvisavla ,yes, toml.go in this version does does still mostly automate this, and would error if UseDB=false and EXPERIMENTAL_BUCKETLIST_DB is present in core config toml.

I added summary on that to prior [comment here](https://github.com/stellar/go/pull/5345#issuecomment-2166703614)

Comment (aditya1702) [reply services/horizon/CHANGELOG.md]: @urvisavla Yes useDB is set to true by default, however there may be some people having left it on false (this actually happened with FT today) and then get an error.

**Review by urvisavla — APPROVED** [review]

**Pull request merged**